### PR TITLE
ENG-80: Implement 'update design' tab

### PR DESCRIPTION
**Pull request by kaf-lamed-beyt**
> [!NOTE]
> Work in progress

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -18,3 +18,7 @@ indent_size = 2
 
 [*.html]
 indent_size = 4
+
+[*.{tsx,ts,js,jsx,md}]
+indent_style = space
+indent_size = 2
```
Comment (AfeezGL) [inline .editorconfig:24]:
```suggestion
```
@kaf-lamed-beyt we are indenting with 4 spaces in all files except .yml files, this ensures consistency across files, `python` / `typescript`.

Comment (kaf-lamed-beyt) [reply .editorconfig]: Awesome! I committed the suggestion now.

File: apps/frontend/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  sudo usermod -aG $USER
+```
+
+- After running this command, you'll need to log out and log back in for the changes to take effect. Alternatively, you can run the following command to apply the changes to your current session:
+
+```shell
+newgrp docker
+```
+
+After following the steps above, you should be able to use the `docker compose` command without the `sudo` flag.
```
Comment (AfeezGL) [inline apps/frontend/CONTRIBUTING.md:43]:
```suggestion
After following the steps above, you should be able to user the `docker compose` command without the `sudo` flag.
```

File: apps/frontend/package.json [hunk 3]
```diff
@@ -17,14 +17,19 @@
     "@radix-ui/react-popover": "^1.0.7",
     "@radix-ui/react-select": "^2.0.0",
     "@radix-ui/react-tabs": "^1.0.4",
+    "@radix-ui/react-tooltip": "^1.0.7",
     "@react-oauth/google": "^0.11.1",
     "@tremor/react": "^3.9.2",
+    "@types/react-color": "^3.0.10",
```
Comment (AfeezGL) [inline apps/frontend/package.json:23]: @kaf-lamed-beyt This should be in dev dependencies.

File: apps/frontend/package.json [hunk 4]
```diff
@@ -17,14 +17,19 @@
...
+    "@types/react-syntax-highlighter": "^15.5.11",
```
Comment (AfeezGL) [inline apps/frontend/package.json:24]: @kaf-lamed-beyt and this too

File: apps/frontend/package.json (hunk 3 above)
Comment (kaf-lamed-beyt) [reply apps/frontend/package.json:23]: Okay. moving it now.

**Comment by kaf-lamed-beyt** [comment]
@AfeezGL, please review

**Review by AfeezGL — APPROVED** [review]

**Pull request merged**